src/main.rs

```diff
@@ -283,7 +283,12 @@ fn main() {
         let working_dir = tmp_path.join(format!("{}", i));
         let source_root = source_root.clone();
 
-        let t = thread::spawn(move || {
+        let mut  thread_name = String::from("Consumer ");
+        thread_name.push_str(&i.to_string());
+
+        let t = thread::Builder::new().name(thread_name);
```

Comment:
Try to see if you can avoid defining the name in a separate variable and just use it here

Reply:
Done!